crates/saya/core/src/data_availability/celestia/mod.rs

```diff
@@ -65,7 +65,27 @@ impl DataAvailabilityClient for CelestiaClient {
         // TODO: we may want to use `blob_get` to ensure the state diff has been published
         // correctly.
         self.client
-            .blob_submit(&[blob], SubmitOptions::default())
+            .blob_submit(&[blob], GasPrice::default())
+            .await
+            .map_err(|e| Error::Client(format!("Celestia RPC error: {e}")))
+    }
+
+    async fn publish_state_diff_and_proof_felts(
+        &self,
+        state_diff: &[FieldElement],
+        state_diff_proof: &[FieldElement],
+    ) -> DataAvailabilityResult<u64> {
+        let bytes: Vec<u8> = state_diff.iter().flat_map(|fe| fe.to_bytes_be().to_vec()).collect();
+        let blob = Blob::new(self.namespace, bytes)?;
+
+        let proof_bytes: Vec<u8> =
+            state_diff_proof.iter().flat_map(|fe| fe.to_bytes_be().to_vec()).collect();
+        let proof_blob = Blob::new(self.namespace, proof_bytes)?;
+
+        // TODO: we may want to use `blob_get` to ensure the state diff has been published
```

Comment:
If you're ok for doing it here, let's do it. If not, let's open an issue to keep track of that in the future.